src/draw_target.rs

```diff
@@ -309,6 +304,62 @@ impl MultiProgressState {
         })
     }
 
+    pub(crate) fn suspend<F: FnOnce() -> R, R>(&mut self, f: F) -> R {
+        self.draw_target
+            .apply_draw_state(ProgressDrawState::new(vec![], true))
+            .ok();
+        let ret = f();
+        let mut lines = vec![];
+        self.gather_draw_state_lines(&mut lines);
+        self.draw_target
+            .apply_draw_state(ProgressDrawState {
+                lines,
+                orphan_lines: 0,
+                force_draw: true,
+                move_cursor: self.move_cursor,
+                alignment: self.alignment,
+            })
+            .ok();
+        ret
+    }
+
+    /// Returns whether any lines were added
+    fn gather_draw_state_lines(&self, lines: &mut Vec<String>) -> bool {
+        let mut will_draw_any_progress_bars = false;
+        for index in self.ordering.iter() {
+            let draw_state = &self.draw_states[*index];
+            if let Some(ref draw_state) = draw_state {
+                will_draw_any_progress_bars = true;
+                lines.extend_from_slice(&draw_state.lines[..]);
+            }
+        }
+        will_draw_any_progress_bars
+    }
+
+    pub(crate) fn println<I: AsRef<str>>(&mut self, msg: I) {
+        let mut lines: Vec<String> = msg.as_ref().lines().map(Into::into).collect();
+        let orphan_lines = lines.len();
+
+        if !self.gather_draw_state_lines(&mut lines) {
+            // The draw state consists entirely of the contents of `msg`. Since
+            // `ProgressDrawState::draw_to_term` doesn't add \n to the last line, we need to add it
+            // ourselves. Otherwise the last line of what we print may be overwritten on next draw.
+            if let Some(line) = lines.last_mut() {
+                line.push('\n');
```

Comment:
is this still needed after https://github.com/console-rs/indicatif/commit/af3bd3154b5f06d3baece6a91eda5dcf48b1e4c0 ?

Reply:
Removing this line and using https://github.com/console-rs/indicatif/pull/356 seems to make it work